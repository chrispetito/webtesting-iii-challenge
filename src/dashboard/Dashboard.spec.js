// Test away
import React from 'react'
import { render } from '@testing-library/react'
import '@testing-library/react/cleanup-after-each'
import Dashboard from './Dashboard'

describe('<Dashboard />', () => {
    it('Dashboard renders without crashing', () => {
        render(<Dashboard />)
    })
    it('dashboard matches snapshot', ()=> {
        const { container } = render(<Dashboard />)
        expect(container).toMatchSnapshot()
    })
    it('renders controls and display', () => {
        const { getByText } = render(<Dashboard />)
        getByText(/^unlocked$/i)
        getByText(/^open$/i)
        getByText(/^lock gate$/i)
        getByText(/^close gate$/i)
    })
})