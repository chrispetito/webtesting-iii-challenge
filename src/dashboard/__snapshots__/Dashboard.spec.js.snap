// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Dashboard /> dashboard matches snapshot 1`] = `
<div>
  <div
    class="display panel"
  >
    <div
      class="led green-led"
      data-testid="locked"
    >
      Unlocked
    </div>
    <div
      class="led green-led"
      data-testid="closed"
    >
      Open
    </div>
  </div>
  <div
    class="controls panel"
  >
    <button
      class="toggle-btn"
      data-testid="locked"
      disabled=""
    >
      Lock Gate
    </button>
    <button
      class="toggle-btn"
      data-testid="closed"
    >
      Close Gate
    </button>
  </div>
</div>
`;
